### PR TITLE
feat(op-chain-ops): WithdrawalNetwork Type

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Pulled out of [a conversation](https://github.com/ethereum-optimism/optimism/pull/6428#discussion_r1275183045) in PR #6428,
this pr implements a `WithdrawalNetwork` type in op-chain-ops genesis that gracefully
handles backwards-compatible parsing of uint8 values and string values.

**Tests**

Unit tests are performed in the genesis package in the config source test file.

**Metadata**

Fixes CLI-4283.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6465** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6465" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6465?utm_source=stack-comment).

File: op-chain-ops/genesis/testdata/test-deploy-config-full.json [hunk 1]
```diff
@@ -44,9 +44,9 @@
   "baseFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
   "l1FeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
   "sequencerFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
-  "baseFeeVaultWithdrawalNetwork": 0,
-  "l1FeeVaultWithdrawalNetwork": 0,
-  "sequencerFeeVaultWithdrawalNetwork": 0,
+  "baseFeeVaultWithdrawalNetwork": "local",
+  "l1FeeVaultWithdrawalNetwork": "local",
+  "sequencerFeeVaultWithdrawalNetwork": "local",
```
Comment (tynes) [inline op-chain-ops/genesis/testdata/test-deploy-config-full.json:49]: can you make one of these be remote so we have coverage of both local and remote from file?

File: op-chain-ops/genesis/testdata/test-deploy-int-nets.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+  "l1StartingBlockTag": "earliest",
+  "l1ChainID": 900,
+  "l2ChainID": 901,
+  "l2BlockTime": 2,
+
+  "maxSequencerDrift": 100,
+  "sequencerWindowSize": 4,
```
Comment (tynes) [inline op-chain-ops/genesis/testdata/test-deploy-int-nets.json]: Any way that we can not introduce this by using a combo of 1/0 and local/remote assertions against existing files?

File: op-chain-ops/genesis/config_test.go [hunk 3]
```diff
@@ -14,6 +14,33 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestWithdrawalNetwork(t *testing.T) {
```
Comment (tynes) [inline op-chain-ops/genesis/config_test.go]: Roundtrip and to uint?

**Review by tynes — APPROVED** [review]

**Pull request merged**